sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -2143,6 +2143,93 @@ public void testHavingOnExactCountDistinct() throws Exception
     );
   }
 
+  @Test
+  public void testExactCountDistinctWithFilter() throws Exception
+  {
+
+    final String sqlQuery = "SELECT COUNT(DISTINCT foo.dim1) FILTER(WHERE foo.cnt = 1), SUM(foo.cnt) FROM druid.foo";
+    // When useApproximateCountDistinct=true and useGroupingSetForExactDistinct=false, planning fails due
+    // to a bug in the Calcite's rule (AggregateExpandDistinctAggregatesRule)
+    try {
+      testQuery(
+          PLANNER_CONFIG_NO_HLL, // Enable exact count distinct
```

Comment:
Even though it is the default behaviour, I think it would be better to use `withOverrides` here too for symmetry with the positive test case.

Reply:
I didn't understand this, did you mean using `withOverrides` with an empty map to ensure symmetry?

Reply:
Understood the original comment, and updated